x/interchainstaking/keeper/ibc_packet_handlers_test.go

```diff
@@ -771,6 +773,168 @@ func (suite *KeeperTestSuite) TestHandleWithdrawForUserLSM() {
+				return &distrtypes.MsgWithdrawValidatorCommission{
+					ValidatorAddress: val,
+				}
+			},
+			triggered: false,
+			err:       true,
+		},
+		{
+			name: "wrong context",
+			setup: func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone) {
+				zone.ConnectionId = ""
+				quicksilver.InterchainstakingKeeper.SetZone(ctx, zone)
+			},
+			checks: func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone) {},
+			msg: func(zone *icstypes.Zone) sdk.Msg {
+				return &distrtypes.MsgWithdrawDelegatorReward{
+					DelegatorAddress: user,
+					ValidatorAddress: val,
+				}
+			},
+			triggered: false,
+			err:       true,
+		},
+		// at this time, the waitgroup will be uint32_max so it's not gonna trigger redemption rate
+		{
+			name: "negative waitgroup",
+			setup: func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone) {
+				zone.WithdrawalWaitgroup = 0
+				quicksilver.InterchainstakingKeeper.SetZone(ctx, zone)
+			},
+			checks: func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone) {},
+			msg: func(zone *icstypes.Zone) sdk.Msg {
+				return &distrtypes.MsgWithdrawDelegatorReward{
+					DelegatorAddress: user,
+					ValidatorAddress: val,
+				}
+			},
+			triggered: false,
+			err:       false,
+		},
+		{
+			name: "valid case with balances != 0",
+			setup: func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone) {
+				zone.WithdrawalWaitgroup = 1
+				balances := sdk.NewCoins(
+					sdk.NewCoin(
+						zone.BaseDenom,
+						math.NewInt(10_000_000),
+					),
+				)
+				zone.DelegationAddress.Balance = balances
+				quicksilver.InterchainstakingKeeper.SetZone(ctx, zone)
+			},
+			checks: func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone) {},
+			msg: func(zone *icstypes.Zone) sdk.Msg {
+				return &distrtypes.MsgWithdrawDelegatorReward{
+					DelegatorAddress: user,
+					ValidatorAddress: val,
+				}
+			},
+			triggered: false,
+			err:       false,
+		},
+		{
+			name: "valid case trigger redemption rate and set zone",
+			setup: func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone) {
+				zone.WithdrawalWaitgroup = 1
+				quicksilver.InterchainstakingKeeper.SetZone(ctx, zone)
+			},
+			checks: func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone) {
+				newZone, found := quicksilver.InterchainstakingKeeper.GetZone(ctx, zone.ChainId)
+				suite.True(found)
+				suite.Zero(newZone.WithdrawalWaitgroup)
+			},
+			msg: func(zone *icstypes.Zone) sdk.Msg {
+				return &distrtypes.MsgWithdrawDelegatorReward{
+					DelegatorAddress: user,
+					ValidatorAddress: val,
+				}
+			},
+			triggered: true,
+			err:       false,
+		},
+		{
+			name: "valid case trigger redemption rate and without set zone",
+			setup: func(ctx sdk.Context, quicksilver *app.Quicksilver, zone *icstypes.Zone) {
+				zone.WithdrawalWaitgroup = 0
+				quicksilver.InterchainstakingKeeper.SetZone(ctx, zone)
```

Comment:
The name of the test case is `name: "valid case trigger redemption rate and without set zone",` but I also see you set zone in line `879`, Is anything wrong here 👀 I think this case is for checking `DelegatorAddress` is `zone.PerformanceAddress.Address` right?

Reply:
This case should exist when we've just registered zone, then the WithdrawalWaitgroup = 0. But in this test we can't setup like that cause the zone need to be get right after suite.setupTestZones(). Do you think we need to separate this case to another test?